libse/IfoParser.cs

```diff
@@ -45,27 +45,27 @@ public List<string> GetAllLanguages()
                     {
                         var ids = SubtitleIDs[i].Split(',');
                         var types = SubtitleTypes[i].Split(',');
-                        if (ids.Length == 2 && ids[0].Trim() == ids[1].Trim() || ids.Length == 3 && ids[0].Trim() == ids[1].Trim() && ids[1].Trim() == ids[2].Trim())
+                        if ((ids.Length == 2 && ids[0].Trim() == ids[1].Trim()) || (ids.Length == 3 && ids[0].Trim() == ids[1].Trim() && ids[1].Trim() == ids[2].Trim()))
                         {
                             list.Add(Subtitles[i] + " (" + ids[0].Trim() + ")");
                         }
                         else
                         {
-                            if (ids.Length >= 1 && types.Length >= 1)
```

Comment:
If we don't evaluate greater expressions first, this will always be true for all the cases.

``` C#
                           if (ids.Length >= 4 && types.Length >= 4)
                            {
                                list.Add(Subtitles[i] + ", " + types[3].Trim() + " (" + ids[3].Trim() + ")");
                            }
                            if (ids.Length >= 3 && types.Length >= 3)
                            {
                                list.Add(Subtitles[i] + ", " + types[2].Trim() + " (" + ids[2].Trim() + ")");
                            }
                            if (ids.Length >= 2 && types.Length >= 2)
                            {
                                list.Add(Subtitles[i] + ", " + types[1].Trim() + " (" + ids[1].Trim() + ")");
                            }
                            if (ids.Length >= 1 && types.Length >= 1)
                            {
                                list.Add(Subtitles[i] + ", " + types[0].Trim() + " (" + ids[0].Trim() + ")");
                            }
```